service/controller/v12/resource/instance/create.go

```diff
@@ -411,59 +414,59 @@ func (r *Resource) deleteDrainerConfig(ctx context.Context, customObject provide
 //     loop 5: worker 2 drained
 //     loop 6: worker 2 reimage
 //
-func (r *Resource) nextInstance(ctx context.Context, customObject providerv1alpha1.AzureConfig, instances []compute.VirtualMachineScaleSetVM, drainerConfigs []corev1alpha1.DrainerConfig, instanceNameFunc func(customObject providerv1alpha1.AzureConfig, instanceID string) (string, error), versionValue map[string]string) (*compute.VirtualMachineScaleSetVM, *compute.VirtualMachineScaleSetVM, *compute.VirtualMachineScaleSetVM, error) {
+func (r *Resource) nextInstance(ctx context.Context, customObject providerv1alpha1.AzureConfig, instances []compute.VirtualMachineScaleSetVM, drainerConfigs []corev1alpha1.DrainerConfig, instanceNameFunc func(customObject providerv1alpha1.AzureConfig, instanceID string) (string, error), versionValue map[string]string) (*workingSet, error) {
 	var err error
 
-	var instanceToUpdate *compute.VirtualMachineScaleSetVM
-	var instanceToDrain *compute.VirtualMachineScaleSetVM
-	var instanceToReimage *compute.VirtualMachineScaleSetVM
+	var ws *workingSet
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the next instance to be updated, drained or reimaged")
 
-		instanceToUpdate, instanceToDrain, instanceToReimage, err = findActionableInstance(customObject, instances, drainerConfigs, instanceNameFunc, versionValue)
+		emptyWorkingSet := workingSet{}
```

Comment:
No need for this.

Reply:
I kinda have preferred to have an explicit difference between the nil case and the emptyWorkingSet, despite the fact that they behave the same way.
But I applied your suggestions anyway.

Reply:
Let's keep that in mind and change it if there are any problems with the current approach. I do also agree that it could confuse sometimes when working with nil (while even valid), but in this case it's not exposed/exported type, it's use is quite narrow and I personally liked the convenience / conciseness of it.

FYI: As an author in general you have right to disagree with reviewer[s] on things like this and proceed with your approach. It's totally fine and happens sometimes. Disagree and commit it is.